test/gateway/eip20_cogateway/revert_redeem.js

```diff
@@ -0,0 +1,363 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const EIP20CoGateway = artifacts.require('TestEIP20CoGateway');
+const BN = require('bn.js');
+const MockUtilityToken = artifacts.require('MockUtilityToken');
+const messageBus = require('../../test_lib/message_bus.js');
+const Utils = require("../../test_lib/utils.js");
+const EventDecoder = require('../../test_lib/event_decoder.js');
+const web3 = require("../../test_lib/web3.js");
+
+const MessageStatusEnum = messageBus.MessageStatusEnum;
+const zeroAddress = Utils.NULL_ADDRESS;
+const zeroBytes = Utils.ZERO_BYTES32;
+
+contract('EIP20CoGateway.revertRedeem()', function (accounts) {
+
+  let constructorParams, redeemParams, utilityToken, eip20CoGateway;
+
+  beforeEach(async function () {
+
+    constructorParams = {
+      valueToken: accounts[9],
+      stateRootProvider: accounts[3],
+      bounty: new BN(100),
+      organization: accounts[9],
+      gateway: accounts[4],
+      burner: zeroAddress,
+      owner: accounts[0],
+    };
+
+    utilityToken = await MockUtilityToken.new(
+      constructorParams.valueToken,
+      "",
+      "",
+      18,
+      constructorParams.organization,
+      { from: constructorParams.owner },
+    );
+
+    constructorParams.utilityToken = utilityToken.address;
+
+    eip20CoGateway = await EIP20CoGateway.new(
+      constructorParams.valueToken,
+      constructorParams.utilityToken,
+      constructorParams.stateRootProvider,
+      constructorParams.bounty,
+      constructorParams.organization,
+      constructorParams.gateway,
+      constructorParams.burner,
+    );
+
+    let hashLockObj = Utils.generateHashLock();
+
+    redeemParams = {
+      amount: new BN(10000),
+      beneficiary: accounts[5],
+      gasPrice: new BN(1),
+      gasLimit: new BN(1000),
+      nonce: new BN(1),
+      hashLock: hashLockObj.l,
+      unlockSecret: hashLockObj,
+      redeemer: accounts[2],
+      intentHash: web3.utils.sha3("dummy"),
+      penalty: constructorParams.bounty.muln(1.5),
+    };
+
+    redeemParams.messageHash = await eip20CoGateway.setMessage.call(
```

Comment:
There is helper method in test_lib/message_bus_utils.js

Reply:
We are using this in other tests too. I think this should be fine.
https://github.com/OpenSTFoundation/mosaic-contracts/blob/develop/test/gateway/eip20_cogateway/progress_redeem.js

`test_lib/message_bus_utils.js` does not exists.

Reply:
used `messageBus.messageDigest`